ts/features/bonus/siciliaVola/components/DestinationSelector.tsx

```diff
@@ -0,0 +1,117 @@
+import * as React from "react";
+import { Dispatch } from "redux";
+import { connect } from "react-redux";
+import { ListItem, View } from "native-base";
+import { debounce } from "lodash";
+import { useContext, useEffect, useState } from "react";
+import { FlatList } from "react-native";
+import { GlobalState } from "../../../../store/reducers/types";
+import TextboxWithSuggestion from "../../../../components/ui/TextboxWithSuggestion";
+import { H4 } from "../../../../components/core/typography/H4";
+import { IndexedById, toArray } from "../../../../store/helpers/indexer";
+import { Municipality, State } from "../types/SvVoucherRequest";
+import { availableMunicipalitiesSelector } from "../store/reducers/voucherGeneration/availableMunicipalities";
+import { svGenerateVoucherAvailableMunicipality } from "../store/actions/voucherGeneration";
+import { LightModalContext } from "../../../../components/ui/LightModal";
+import I18n from "../../../../i18n";
+import WrappedFlatlist from "./WrappedMunicipalityFlatlist";
+
+type OwnProps = {
+  availableStates: IndexedById<State>;
```

Comment:
Since this component will be used within screens, I have foreseen that the user will not be able to access those screens if the states cannot be loaded.
This way I can show an error page with a try again button.
Does this seem like an acceptable solution?